streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/OnErrorResumeNextProcessor.kt

```diff
@@ -0,0 +1,59 @@
+package com.mirego.trikot.streams.reactive.processors
+
+import com.mirego.trikot.foundation.concurrent.AtomicReference
+import com.mirego.trikot.foundation.concurrent.dispatchQueue.SynchronousSerialQueue
+import com.mirego.trikot.streams.cancellable.CancellableManagerProvider
+import com.mirego.trikot.streams.reactive.StreamsProcessorException
+import com.mirego.trikot.streams.reactive.observeOn
+import com.mirego.trikot.streams.reactive.subscribe
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+
+typealias OnErrorResumeNextBlock<T> = (Throwable) -> Publisher<T>
+
+class OnErrorResumeNextProcessor<T>(parentPublisher: Publisher<T>, private var block: OnErrorResumeNextBlock<T>) :
+    AbstractProcessor<T, T>(parentPublisher) {
+
+    override fun createSubscription(subscriber: Subscriber<in T>): ProcessorSubscription<T, T> {
+        return OnErrorResumeNextProcessorSubscription(subscriber, block)
+    }
+
+    class OnErrorResumeNextProcessorSubscription<T>(
+        private val subscriber: Subscriber<in T>,
+        private val block: OnErrorResumeNextBlock<T>
+    ) : ProcessorSubscription<T, T>(subscriber) {
+        private val cancellableManagerProvider = CancellableManagerProvider()
+        private val onErrorValidation = AtomicReference(0)
+        private val serialQueue = SynchronousSerialQueue()
+
+        override fun onCancel(s: Subscription) {
+            super.onCancel(s)
+            cancellableManagerProvider.cancel()
+        }
+
+        override fun onNext(t: T, subscriber: Subscriber<in T>) {
+            subscriber.onNext(t)
+        }
+
+        override fun onError(t: Throwable) {
+            onErrorValidation.setOrThrow(0, 1)
```

Comment:
onErrorValidation should never be set to 0 afterward as onError is a terminal event.

I would suggest
```
if (!onErrorValidation.compareAndSet(0, 1)) {
  throw IllegalStateException("A cool error message explaining that its not normal to revieve onError twice in the same processor")
}
```

Reply:
onErrorValidation.setOrThrow(0, 1) does that,